docs/configuration.md

```diff
@@ -1118,6 +1118,7 @@ is unsupported.
 | `remoteurl`| yes     | The URL for the repository on Docker Hub.             |
 | `username` | no      | The username registered with Docker Hub which has access to the repository. |
 | `password` | no      | The password used to authenticate to Docker Hub using the username specified in `username`. |
+| `ttl`      | no      | Expire proxy cache configured in "storage" after this time. Cache expiration is disabled if not provided.  |
```

Comment:
Should we keep the default same as the current value? (7 days)? I that case, possibly use some special value for disabling

Reply:
In this PR implementation, if no config is specified there is no expiration. I like this approach as it forces explicit intent by the operator, rather than hiding some opaque behaviour. As long as the behaviour is documented and we no longer pick arbitrary values I'm happy with this.

Reply:
One thing to note is that operators who might be used to seeing reduced space will see it growing unbounded after this PR is deployed. Considering most of the operators would've encountered this bug after 7 days I think its fine to have this backward incompatible change.

Reply:
If no default value set here, does it impact the upgrade scenario? The default expiration setting will change to none? We probably should document this change. @milosgajdos

Reply:
So this entry is used for generating the docs, actually. How we _notify/warn_ users about this change is a whole another question. But I can see this turning a bit tricky. I had a read through related issues [here](https://github.com/distribution/distribution/issues/2367) and [here](https://github.com/distribution/distribution/issues/3247) and it seems that people either:
1. Don't set `STORAGE_CACHE_BLOBDESCRIPTOR` when local file store is used
2. Build a custom image and set expiry `TTL` to a very large number

Reply:
I think we should make it default to 7 days, otherwise, it's a breaking change IMO. People using this in production have set up their environments according to the default behavior, changing that without a major release is a risk IMO.

Are there any downsides in making it default to 7 days? Anyone that's not happy with that can override it, and we could use a value of `-1` to disable it.

Reply:
> Are there any downsides in making it default to 7 days?

I see that "disabled by default" can mitigate #2367, but anyone facing that issue could still do it with `ttl: -1`.

Reply:
> Anyone that's not happy with that can override it, and we could use a value of -1 to disable it.

Curious if we could make the type for the TTL a pointer instead, we should be able to tell 0 from not configured with a nil check then and not have to have special values.

Reply:
@manishtomar WDYT?

Reply:
Since the default behavior is effectively broken (in a way that is difficult to debug) I feel changing the default behavior is the right thing to do. We are still making operators life difficult by silently failing to pull images after 7 days. Currently everyone who encounters the bug has to rebuild the code with high TTL and deploy it. The only thing this PR will do is avoid rebuilding from source. They will _still_ have to face the error, google it and disable TTL to use it. Considering that the end state is always disabling TTL to make the system work why not make it default? FWIW while it seems bad to make backward-incompatible operator changes we've historically [done that in the past](https://github.com/distribution/distribution/issues/2367#issuecomment-454805687) with minor version upgrades. That is not a justification to make another one but seems justifiable in this case.

Reply:
Thanks, @manishtomar o/ I think we all agree here re: the default behavior being broken and our having to change it. The "issue" is not so much the change of the current behavior on `main` branch, which no one would object to going forward, but rather that we'd like to propagate this change into the next `2.7.x` patch release and would like to avoid changing the existing behavior, rather than tracking another patch. Would you consider changing this PR to what @SteveLasker suggests in his [comment ](https://github.com/distribution/distribution/pull/3238#issuecomment-827660735)?

Reply:
Ping @manishtomar